### PR TITLE
Test library api

### PR DESCRIPTION
**Pull request by alteredco**
have connected the sessions/{token} api up to the test library page. Currently have a default token set in the test library page to show some data. Expecting token to be accessible as context.query.token once the token url is set up.

### CODE REVIEW

File: components/TestLibraryStepper/TestLibraryStepper.tsx [hunk 1]
```diff
@@ -1,41 +1,46 @@
...
-
+function getSteps(testArr: CandidateTestStatus[]) {
     //set labels for steps
     const testLabelArray = [];
-    while (testLabelArray.length + 1 <= testNumber) {
-        testLabelArray.push(`Test ${testLabelArray.length + 1}`);
-    }
+    testLabelArray.push(testArr.map(test => test.testName));
     return testLabelArray;
 }
```
Comment (MichaelWalker) [inline components/TestLibraryStepper/TestLibraryStepper.tsx:23]: I think this whole method can now just be 
```ts
return testArray.map(test => test.testName);
```

File: components/TestLibraryStepper/TestLibraryStepper.tsx [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
 
+function getActiveStep(testArr: CandidateTestStatus[]) {
```
Comment (MichaelWalker) [inline components/TestLibraryStepper/TestLibraryStepper.tsx:25]: We should probably give this (and the other functions) return types.

File: components/TestLibraryStepper/TestLibraryStepper.tsx [hunk 3]
```diff
@@ -1,41 +1,46 @@
...
     return (
         <article className="stepperContainer">
             <MuiThemeProvider theme={TestSwitchTheme}>
                 <Stepper style={{backgroundColor: "transparent"}} alternativeLabel activeStep={activeStep}
-                         connector={<TestSwitchConnector/>}>
+                         connector
+                             ={<TestSwitchConnector/>}>
                     {steps.map((label, status) => (
-                        <Step key={label}>
+                        <Step key={label.length}>
```
Comment (MichaelWalker) [inline components/TestLibraryStepper/TestLibraryStepper.tsx:43]: label.length seems odd - is that deliberate?

File: pages/api/candidateApiClient.module.ts [hunk 4]
```diff
@@ -1,36 +1,24 @@
 ﻿import fetch from "node-fetch";
```
Comment (MichaelWalker) [inline pages/api/candidateApiClient.module.ts:1]: Would you mind moving this out of the `pages` directory?
Its not really a page, so fine to just have a `/api` directory at the root level.

File: pages/api/candidateApiClient.module.ts [hunk 5]
```diff
@@ -1,36 +1,24 @@
...
-        return error.message;
-    }
+export interface CandidateTestStatus {
+    testName: string;
+    testStatus: string;
 }
 
-export async function getCandidateTests() {
+export async function getSessionCandidate(token: string): Promise<SessionCandidate> {
+    const baseUrl = `https://testswitch-api-staging.herokuapp.com/sessions/`;
```
Comment (MichaelWalker) [inline pages/api/candidateApiClient.module.ts:15]: Could this be configurable from an environment variable?

File: pages/api/candidateApiClient.module.ts [hunk 6]
```diff
@@ -1,36 +1,24 @@
...
     try {
         const result = await fetch(
-            //TODO: placeholder endpoint
-            `https://testswitch-api-staging.herokuapp.com/candidates`
+            `${baseUrl}${token}`
```
Comment (MichaelWalker) [inline pages/api/candidateApiClient.module.ts:18]: I _think_ the base URL probably shouldn't include `sessions`.
Could we change this to

`${baseUrl}/sessions/${token}`?

File: pages/testlibrary.tsx [hunk 7]
```diff
@@ -20,24 +20,21 @@ const TestLibrary: NextPage<TestlibraryProps> = ({candidateTestStatus}) => {
...
-    return 1;
-};
+//TODO placeholder id for candidateID,until url has token as query
+const candidateToken = `815b47a8-0b7a-4d6a-99e9-2fe130c5b774`;
 
 export const getServerSideProps: GetServerSideProps = async context => {
-    const tests = getCandidateTests();
-    const results = getCandidateTestResults();
+    const token = context.query.token as string;
+    const sessionData = getSessionCandidate(candidateToken);
```
Comment (MichaelWalker) [inline pages/testlibrary.tsx:33]: could we use `token` here rather than `candidateToken`?

Comment (alteredco) [reply pages/testlibrary.tsx]: yep. just for testing. Now cleared out. ✔️

File: pages/api/candidateApiClient.module.ts (hunk 6 above)
Comment (alteredco) [reply pages/api/candidateApiClient.module.ts]: done. ✔️

File: pages/api/candidateApiClient.module.ts (hunk 4 above)
Comment (alteredco) [reply pages/api/candidateApiClient.module.ts]: yep, no problem, it was just what they did in the Nextjs documentation so I blindly followed that. changed now 👍

File: components/TestLibraryStepper/TestLibraryStepper.tsx (hunk 3 above)
Comment (alteredco) [reply components/TestLibraryStepper/TestLibraryStepper.tsx]: oops! changed to index as key. ✔️

File: components/TestLibraryStepper/TestLibraryStepper.tsx (hunk 2 above)
Comment (alteredco) [reply components/TestLibraryStepper/TestLibraryStepper.tsx]: agreed ✔️

File: components/TestLibraryStepper/TestLibraryStepper.tsx (hunk 1 above)
Comment (alteredco) [reply components/TestLibraryStepper/TestLibraryStepper.tsx]: of course. done! ✔️

**Pull request merged**